pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -555,6 +587,13 @@ public boolean commitSegmentMetadata(String rawTableName, final String committin
     ZNRecord newZnRecord =
         makeZnRecordForNewSegment(rawTableName, newInstances.size(), newStartOffset, newSegmentNameStr);
     final LLCRealtimeSegmentZKMetadata newSegmentZKMetadata = new LLCRealtimeSegmentZKMetadata(newZnRecord);
+
+    // Add the partition metadata if available.
+    SegmentPartitionMetadata partitionMetadata = getColumnMetadata(realtimeTableName, partitionId);
```

Comment:
We should limit the calls to get table metadata multiple times during a segment commit. Can we cache this?